### ISSUE TITLE
Fix indentation check for continuations of commands starting on same line as script

### ISSUE THREAD

**Issue by nmoroze** [nmoroze/tclint#35]
Examples:

```tcl
if {1} {puts $chan \
    "foo"}
```

Actual:
```
test.tcl:2:1: expected indent of 0 spaces, got 4 [indent]
```

Expected: no violations

```tcl
eval cmd \
a \
    b
```

Actual:
```
test.tcl:3:1: expected indent of 0 spaces, got 4 [indent]
```

Expected:
```
test.tcl:2:1: expected indent of 4 spaces, got 0 [indent]
```

**Comment by maliberty** [comment]
I'm seeing the same problem in OR (eg https://github.com/The-OpenROAD-Project/OpenROAD/actions/runs/9783206598/job/27011254166?pr=5272) 
![image](https://github.com/nmoroze/tclint/assets/761514/451b8c34-875e-4fd4-bb3e-ec1b2d79bd09)

![image](https://github.com/nmoroze/tclint/assets/761514/03e42af8-1059-45a2-be16-b135163d5584)

**Comment by nmoroze** [comment]
@maliberty thanks for the pointer, I'll take a look ASAP.

In the meantime, you can add comments to locally disable indent checking and get your CI back to green. 

```tcl
# tclint-disable indent
[ code ]
# tclint-enable indent
```

See [docs](https://github.com/nmoroze/tclint/blob/main/docs/configuration.md#ignore-violations-with-inline-comments) for more info. 

BTW, I think tclint will want a different style than you have, e.g. it'll ask for: 

```tcl
mpl2::set_debug_cmd $block \
  $coarse \
  $ fine \
  ...
```

If you prefer the aligned style I can look into supporting it, but I think that'll be lower priority.

**Comment by maliberty** [comment]
We have lots of other code like this that doesn't get flagged.  Any idea why just here?

**Comment by nmoroze** [comment]
The violation on line 313 is related to the `# checker` comment. Dropping the closing brace to a lower line seems to be how this is done in other parts of the codebase, e.g. https://github.com/The-OpenROAD-Project/OpenROAD/blob/efe05a9c9aee2955753945eca8a9e7aaee40b8cd/src/pad/src/pad.tcl#L142. That fixes the problem, although this is a bug in tclint that I'll address.

I think the remaining violations are actually just tclint working as intended. Do you have any examples of it allowing that style elsewhere? There are two exceptions where it won't check the indentation of multiline commands (in expressions [#20] and command substitutions [#23]), so it's possible the other examples you've seen have been one of those two cases (e.g. the `mpl2::rtl_macro_placer_cmd` in this file is in an if statement expression, so it's not currently being checked).

**Comment by maliberty** [comment]
Perhaps it is just that those two exceptions dominate.  I would like to see the indented style supported.  For now I've used your disable suggestion.

**Comment by nmoroze** [comment]
I'm closing this now that the latest tclint release has removed indentation checks (in favor of using tclfmt).

There's still the matter of the indented style - I made the decision not to support this in tclfmt, but if there's still interest I could open a new issue to track it (but it'll likely be a low priority).
